src/preamble.js

```diff
@@ -46,12 +46,33 @@ function getSafeHeapType(bytes, isFloat) {
 var SAFE_HEAP_COUNTER = 0;
 #endif
 
+var safe_heap_warn_set = new Set();
+
 function SAFE_HEAP_STORE(dest, value, bytes, isFloat) {
 #if SAFE_HEAP_LOG
   Module.print('SAFE_HEAP store: ' + [dest, value, bytes, isFloat, SAFE_HEAP_COUNTER++]);
 #endif
   if (dest <= 0) abort('segmentation fault storing ' + bytes + ' bytes to address ' + dest);
-  if (dest % bytes !== 0) abort('alignment error storing to address ' + dest + ', which was expected to be aligned to a multiple of ' + bytes);
+  if (dest % bytes !== 0)
```

Comment:
Ooh, this is actually a great feature: don't fail on alignment failure, but (optionally) print it out and keep on trucking. We definitely would like to have this feature even for Wasm!

The motivation is that unaligned accesses are slower than aligned ones on both x86 and ARM (https://stackoverflow.com/questions/12491578/whats-the-actual-effect-of-successful-unaligned-accesses-on-x86, http://infocenter.arm.com/help/index.jsp?topic=/com.arm.doc.faqs/ka15414.html), so performance sensitive code may be interested in profiling where these occur even in Wasm code.

Though to make this stick, I think we want to:

a) it looks like if the way this is currently implemented, when it an alignment fault is detected, this will warn but still perform the incorrectly aligned memory operation. Would you be able to improve this so that on a detected misaligned access, this would fix up the alignment so that the result will run correctly?

b) Could we change the code compilation options to something like this:

- `-s SAFE_HEAP=0`: Require correct alignment in asm.js
- `-s SAFE_HEAP=1`: Detect alignment faults, and abort on first occurrence. (the old way these functions used to be)
- `-s SAFE_HEAP=2`: Detect alignment faults, but warn when they occur, fix up, and continue (the new way)

Then `-s UNALIGNED_ACCESS_FUNCTION_LIST=...` and `-s SAFE_HEAP=2` could then be treated orthogonally. The different `-s SAFE_HEAP=1/2` modes can be implemented with `#if SAFE_HEAP == 2` types of preprocessor checks in the `src/preamble.js`.